references/classification/train.py

```diff
@@ -186,6 +182,10 @@ def main(args):
         model, optimizer = amp.initialize(model, optimizer,
```

Comment:
Does this change the optimizer in a way that breaks `lr_scheduler` (i.e., it points to a different optimizer and thus don't update the optimizer lr properly)?

Reply:
very good question. I don't have a definitive answer, but maybe a simple approach is just to swap the order of these:
```
    lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=args.lr_step_size, gamma=args.lr_gamma)

    if args.apex:
        model, optimizer = amp.initialize(model, optimizer,
                                          opt_level=args.apex_opt_level
                                          )
```
thoughts?

Reply:
With the current code order, at epoch 190, I've got this:
```Epoch: [195]  [   0/1252]  eta: 2:27:28  lr: 0.0008934519439965433  img/s: 755.8400387443562  loss: 0.6304 (0.6304)  acc1: 86.7188 (86.7188)  acc5: 94.5312 (94.5312)  time: 7.0678  data: 6.6480  max mem: 14467
```
Looks like the learning rate is behaving just fine.

Reply:
The `amp.initialize` [documentation](https://nvidia.github.io/apex/amp.html) does seem to indicate that the returned optimizer is the same object that was passed as an argument, but I do not see any reason to assume this is true or will continue to be true. As @fmassa suggested, we could just define the scheduler after the call to `amp.initialize`.

Reply:
agreed